### ISSUE TITLE
Button Component!

### ISSUE THREAD

**Issue by waldnercharles** [HealthCatalyst/Fabric.Cashmere#4]
Should we change the structure of our button component? I kind of like the idea of having a component that uses an attribute selector.

Thoughts?

**Comment by jtrujill** [comment]
I like the idea of having a component which uses a selector of:

```
@Component({
selector: 'button[hc-button], a[hc-button]'
})
class HCButton {
}
```

This will allow us to stylize buttons and also links that have the hc-button attribute.

**Comment by waldnercharles** [comment]
To expand the conversation further, and potentially cause some controversy we could even do something like this:
```
@Component({
  selector: 'button[hc-button], a[hc-button]'
})
class HCButton {
  @Input('hc-button') props: ButtonProps; // Merge me with the default props below!
  private defaultProps: ButtonProps;
}
```

This would let us consume our button doing something like: `<button [hc-button]="coolButtonProps">`. This is similar to the style @corytak has implemented where all of the button's input properties are in a single ButtonOptions class. Though, that is also up for debate.

**Comment by benjanderson** [comment]
many of the requested changes are implemented in #6
